components/Search.js

```diff
@@ -24,16 +24,33 @@ function fetchData () {
 function Search () {
   // eslint-disable-next-line no-unused-vars
   const [destination, setDestination] = React.useState("");
+  const [buildingName, setBuildingName] = React.useState("");
   const [data, setData] = React.useState();
 
+  let destinationName = destination.name;
+
+  if (destinationName === undefined) {
```

Comment:
Why don't you put this inside `destinationSetter`?

```
function destinationSetter (destination) {
    // resets the value of the buildingName label when on pressing on a searched item
    setDestination(destination);
    setBuildingName("");
    AsyncStorage.setItem("destination", destination.name.toString());
    AsyncStorage.setItem("buildingSelected", buildingName.toString());
  }
```

Reply:
This whole component needs refactoring. It is **not** a **reusable** component in this state. That's why the **doubleSearch** functional component is rewriting this component twice...

Reply:
@AlainJobs Agreed. We will make the appropriate changes in the refactoring.